components/raftstore-v2/src/worker/engine_gc.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{
+    error::Error as StdError,
+    fmt::{self, Display, Formatter},
+    sync::mpsc::Sender,
+    time::Duration,
+};
+
+use engine_traits::{RaftEngine, RaftLogGcTask};
+use file_system::{IoType, WithIoType};
+use slog::{error, Logger};
+use thiserror::Error;
+use tikv_util::worker::{Runnable, RunnableWithTimer};
+
+const MAX_GC_REGION_BATCH: usize = 512;
+
+pub enum Task {
+    RaftLogGc {
+        region_id: u64,
+        start_idx: u64,
+        end_idx: u64,
+    },
+}
+
+impl Task {
+    pub fn raft_log_gc(region_id: u64, start_idx: u64, end_idx: u64) -> Self {
```

Comment:
Because we always use `RaftLogEngine` for v2, so it seems we don't need a worker at all. The logs can be just deleted in async writer a long side with other metas.